src/bucket.js

```diff
@@ -145,8 +163,15 @@ export default class Bucket {
    * @return {Promise<Object, Error>}
    */
   deleteCollection(collection, options={}) {
+    const collectionObj = toDataBody(collection);
+    if (!collectionObj.id) {
+      throw new Error("A collection id is required.");
+    }
+
+    options = { last_modified: collectionObj.last_modified, ...options };
```

Comment:
It's probably worth destructuring `collectionObj` above, eg. `const {id, last_modified} = collectionObj;`